<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1151</width>
    <height>597</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Course work Yermolenko Denys KV-73</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTabWidget" name="maniplationTab">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="graphEditorTab">
        <attribute name="title">
         <string>Edit Graph</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="networkDegreeLabel">
           <property name="text">
            <string>NetworkDegree</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="networkDegreeValue">
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Node</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Id:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="nodeIdLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="1" column="0" colspan="2">
               <widget class="QCheckBox" name="isHostCheckBox">
                <property name="text">
                 <string>Is Host</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_2">
             <property name="title">
              <string>Line</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="2" column="1">
               <widget class="QLineEdit" name="error_chanceLineEdit"/>
              </item>
              <item row="0" column="2">
               <widget class="QComboBox" name="lineWeightComboButton">
                <item>
                 <property name="text">
                  <string>1</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>3</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>5</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>6</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>8</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>10</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>11</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>15</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>18</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>20</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>22</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>26</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>27</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>ErrorChance</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0" colspan="2">
               <widget class="QCheckBox" name="isHalfDuplex">
                <property name="text">
                 <string>HalfDuplex</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="2">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Weight</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Line weight during generation</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="autoWeightRadioButton">
             <property name="text">
              <string>auto</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="manualWeightRadioButton">
             <property name="text">
              <string>manual</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="manualWeightComboBox">
             <item>
              <property name="text">
               <string>1</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>3</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>5</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>6</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>8</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>10</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>11</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>15</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>18</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>20</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>22</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>26</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>27</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="sendTab">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <attribute name="title">
         <string>Send Message</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="2" column="3">
          <widget class="QComboBox" name="station2ComboBox"/>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Header Size</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="station1ComboBox"/>
         </item>
         <item row="0" column="0" colspan="4">
          <widget class="QPushButton" name="startSendingButton">
           <property name="text">
            <string>Start sending</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Station 2</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="4">
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>Type of sending</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QRadioButton" name="datagramRadioButton">
              <property name="text">
               <string>Datagram</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="logConnectionRadioButton">
              <property name="text">
               <string>Logical Connection</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="virtChannelRadioButton">
              <property name="text">
               <string>Virtual channel</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>PackageSize</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Station 1</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1" colspan="3">
          <widget class="QLineEdit" name="packageSizeLineEdit">
           <property name="text">
            <string>1200</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1" colspan="3">
          <widget class="QLineEdit" name="headerSizeLineEdit">
           <property name="text">
            <string>50</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1" colspan="3">
          <widget class="QLineEdit" name="messageSizeLineEdit">
           <property name="text">
            <string>3000</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Message size</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0" colspan="4">
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>Simulation option</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QRadioButton" name="realtimeRadioButton">
              <property name="text">
               <string>realtime</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="fastResultButton">
              <property name="text">
               <string>fastResult</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="MainView" name="graphicsView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>2</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>600</width>
         <height>0</height>
        </size>
       </property>
       <property name="sceneRect">
        <rectf>
         <x>0.000000000000000</x>
         <y>0.000000000000000</y>
         <width>1000.000000000000000</width>
         <height>1000.000000000000000</height>
        </rectf>
       </property>
       <property name="renderHints">
        <set>QPainter::HighQualityAntialiasing|QPainter::TextAntialiasing</set>
       </property>
      </widget>
      <widget class="QTabWidget" name="tabWidget_2">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="nodeAndLineDetails">
        <attribute name="title">
         <string>RoutingAlgorithm</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="1" column="0">
          <widget class="QGroupBox" name="routingTableConfGroupBox">
           <property name="title">
            <string>RoutingTables configuraiton</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="2" column="0">
             <widget class="QPushButton" name="nextStepButton">
              <property name="text">
               <string>NextStep</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="stepCounterLabel">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QPushButton" name="startRoutingAlgoButton">
                <property name="text">
                 <string>Start Vector-distance algo</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="ByWeightsRadioButton">
                <property name="text">
                 <string>ByWeights</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="ByNodesRadioButton">
                <property name="text">
                 <string>ByNodes</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="3" column="0">
             <widget class="QPushButton" name="endConfigurationButton">
              <property name="text">
               <string>EndConfiguration</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QPushButton" name="enableEditModeButton">
           <property name="text">
            <string>Enable Edit Mode</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QGroupBox" name="showPathsGroupBox">
           <property name="title">
            <string>Show paths</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="6" column="0" colspan="4">
             <widget class="QListView" name="pathsListView"/>
            </item>
            <item row="7" column="0" colspan="3">
             <widget class="QTextEdit" name="textPathReprez"/>
            </item>
            <item row="5" column="0" colspan="4">
             <widget class="QPushButton" name="getShortestPathsButton">
              <property name="text">
               <string>GetShortestPaths</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="node1Label">
              <property name="text">
               <string>Station1</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="node2Label">
              <property name="text">
               <string>Station 2</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="node1ComboBox"/>
            </item>
            <item row="0" column="3">
             <widget class="QComboBox" name="node2ComboBox"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>Tab 2</string>
        </attribute>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1151</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuScene">
    <property name="title">
     <string>Scene</string>
    </property>
    <addaction name="actionZoom"/>
    <addaction name="actionZoom_2"/>
    <addaction name="separator"/>
    <addaction name="actionSelection"/>
    <addaction name="actionMove"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuScene"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionZoom"/>
   <addaction name="actionZoom_2"/>
   <addaction name="actionMove"/>
   <addaction name="actionSelection"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionZoom">
   <property name="text">
    <string>Zoom+</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="actionZoom_2">
   <property name="text">
    <string>Zoom-</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionMove">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Move</string>
   </property>
  </action>
  <action name="actionSelection">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Selection</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QGraphicsView</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
